app/src/main/java/com/example/mycalculator/mvp/model/CalculatorModel.java

```diff
@@ -2,16 +2,75 @@
 
 import com.example.mycalculator.mvp.contract.CalculatorContract;
 
+import static com.example.mycalculator.utils.ConstantsUtils.OPERATOR_DIVIDE;
+import static com.example.mycalculator.utils.ConstantsUtils.OPERATOR_MINUS;
+import static com.example.mycalculator.utils.ConstantsUtils.OPERATOR_MULTIPLY;
+import static com.example.mycalculator.utils.ConstantsUtils.OPERATOR_SUM;
+
+import static com.example.mycalculator.utils.StringUtils.EMPTY_STRING;
+import static com.example.mycalculator.utils.StringUtils.ERROR_MESSAGE;
+
 public class CalculatorModel implements CalculatorContract.CalculatorModelContract {
 
-    private String firstOperand;
+    private String firstOperand = EMPTY_STRING;
+    private String secondOperand = EMPTY_STRING;
+    private String operator = EMPTY_STRING;
+    private String result = EMPTY_STRING;
+
 
     @Override
     public void saveNumber(String number) {
-        firstOperand = number;
+        if (operator.isEmpty()) {
+            firstOperand += number;
+        } else {
+            secondOperand += number;
+        }
+    }
+    @Override
+    public String getPartialResult() {
+
+        return (firstOperand + operator + secondOperand);
+    }
+
+    @Override
+    public String getFullResult() {
+
+        switch (operator) {
+
+            case OPERATOR_SUM:
+                result = String.valueOf((Double.parseDouble(firstOperand) + Double.parseDouble(secondOperand)));
+                break;
+
+            case OPERATOR_MINUS:
+                result = String.valueOf((Double.parseDouble(firstOperand) - Double.parseDouble(secondOperand)));
+                break;
+
+            case OPERATOR_DIVIDE:
+                result = String.valueOf((Double.parseDouble(firstOperand) / Double.parseDouble(secondOperand)));
+                break;
+
+            case OPERATOR_MULTIPLY:
+                result = String.valueOf((Double.parseDouble(firstOperand) * Double.parseDouble(secondOperand)));
+                break;
+
+            default:
+                result = ERROR_MESSAGE;
+        }
+
+        return result;
     }
 
-    public String getResult() {
-        return firstOperand;
+    @Override
+    public String eraseResult() {
+        result = EMPTY_STRING;
+        firstOperand = EMPTY_STRING;
+        secondOperand = EMPTY_STRING;
+        operator = EMPTY_STRING;
+        return EMPTY_STRING;
+    }
+
+    @Override
+    public void saveOperationSymbol(String toString) {
```

Comment:
use a more representative name on the parameter's variable

Reply:
Done